mono/metadata/cominterop.c

```diff
@@ -3756,3 +3756,26 @@ mono_cominterop_get_com_interface_internal (gboolean icall, MonoObjectHandle obj
 	g_assert_not_reached ();
 #endif
 }
+
+gboolean
+mono_cominterop_is_interface (MonoClass* klass)
+{
+#ifndef DISABLE_COM
+	ERROR_DECL (error);
+	MonoCustomAttrInfo* cinfo = NULL;
+	MonoInterfaceTypeAttribute* itf_attr = NULL;
+
+	cinfo = mono_custom_attrs_from_class_checked (klass, error);
+	mono_error_assert_ok (error);
+	if (cinfo) {
+		itf_attr = (MonoInterfaceTypeAttribute*)mono_custom_attrs_get_attr_checked (cinfo, mono_class_get_interface_type_attribute_class (), error);
+		mono_error_assert_ok (error); /*FIXME proper error handling*/
```

Comment:
What do we want proper error handling to be?  Can we just return false if trying to look for the custom attribute fails?  In that case could do something like this:

```c
if (!is_ok (error)) {
    mono_error_cleanup (error);
    itf_attr = NULL;
}
```

Reply:
I essentially just copy-pasted that part from cominterop_get_com_slot_begin (admittedly, I should have at least separated it out into a common helper). I wasn't really sure what proper error handling would be, but that seems reasonable enough. I'll resend the patch.